File: Examples2018/BcastRecCompOrdBcastWithResRec/app/src/main/java/course/examples/broadcastreceiver/resultreceiver/Receiver1.java
package course.examples.broadcastreceiver.resultreceiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

public class Receiver1 extends BroadcastReceiver {
	
	private final String TAG = "Receiver1";

	@Override
	public void onReceive(Context context, Intent intent) {
		
		Log.i(TAG, "INTENT RECEIVED by Receiver1");

		String tmp = getResultData() == null ? "" : getResultData();
		setResultData(tmp + ":Receiver 1:");
	}

}

File: ExamplesKotlin/BcastRecCompOrdBcastWithResRec/app/src/main/java/course/examples/broadcastreceiver/resultreceiver/Receiver1.kt
package course.examples.broadcastreceiver.resultreceiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log

class Receiver1 : BroadcastReceiver() {

    companion object {
        private const val TAG = "Receiver1"
    }

    override fun onReceive(context: Context, intent: Intent) {

        Log.i(TAG, "INTENT RECEIVED by Receiver1")

        val tmp = if (resultData == null) "" else resultData
        resultData = "$tmp:Receiver 1:"
    }

}
